walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,57 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.PriorityQueue;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<String> attendees = request.getAttendees();
+    int initialQueueCapacity = 1;
+    if (events.size() > 0) {
+      initialQueueCapacity = events.size();
+    }
+    PriorityQueue<TimeRange> blockingIntervals =
+        new PriorityQueue<>(initialQueueCapacity, TimeRange.ORDER_BY_START);
+    for (Event event : events) {
+      if (eventContainsAttendee(event, attendees)) {
+        blockingIntervals.add(event.getWhen());
+      }
+    }
+    ArrayList<TimeRange> freeSlots = new ArrayList<>();
+    long duration = request.getDuration();
+    if (duration < TimeRange.WHOLE_DAY.duration()) {
+      if (!blockingIntervals.isEmpty()) {
+        int previousEnd = TimeRange.START_OF_DAY;
+        while (!blockingIntervals.isEmpty()) {
+          TimeRange timeRange = blockingIntervals.remove();
+          int intervalDifference = previousEnd - timeRange.start();
```

Comment:
Why not timeRange.start() - previousEnd and then just check that it's  >= duration?